castle/cms/passwordvalidation/hibp.py

```diff
@@ -0,0 +1,49 @@
+from castle.cms.interfaces.passwordvalidation import ICustomPasswordValidator
+from plone import api
+from zope.interface import implementer
+import hashlib
+import urllib2
+
+from castle.cms import _
+
+
+@implementer(ICustomPasswordValidator)
+class HaveIBeenPwned(object):
+    """ Checks if the password has already been used
+    by iterating over the memberdata property password_history_size.
+    """
+
+    def __init__(self, context):
+        self.context = context
+
+    def validate(self, password, data):
+        # Permit empty passwords here to allow registration links.
+        # Plone will force the user to set one.
+        if password is None:
+            return None
+
+        if api.user.is_anonymous():
+            # No check for registrations.
+            return None
+
+        hibp_enabled = api.portal.get_registry_record(
```

Comment:
Can provide `default=False`? Otherwise, this could throw an error in the period where the upgrade step hasn't been run yet.

Reply:
Done.